std/src/main/scala/algebra/std/set.scala

```diff
@@ -1,17 +1,31 @@
 package algebra
 package std
 
+import algebra.lattice.Lattice
 import algebra.ring.Semiring
 
 package object set extends SetInstances
 
 trait SetInstances {
-  implicit def setEq[A]: Eq[Set[A]] = new SetEq[A]
+  implicit def setLattice[A]: Lattice[Set[A]] = new SetLattice[A]
+  implicit def setPartialOrder[A]: PartialOrder[Set[A]] = new SetPartialOrder[A]
   implicit def setSemiring[A] = new SetSemiring[A]
 }
 
-class SetEq[A] extends Eq[Set[A]] {
-  def eqv(x: Set[A], y: Set[A]): Boolean = x == y
+class SetLattice[A] extends Lattice[Set[A]] {
+  def join(lhs: Set[A], rhs: Set[A]): Set[A] = lhs.union(rhs)
+
+  def meet(lhs: Set[A], rhs: Set[A]): Set[A] = lhs.intersect(rhs)
+}
+
+class SetPartialOrder[A] extends PartialOrder[Set[A]] {
```

Comment:
Would you consider something that makes fewer passes over the sets? For example:

```scala
def partialCompare(x: Set[A], y: Set[A]): Double =
  if (x.size < y.size) (if (x.forall(y)) -1.0 else Double.NaN)
  else if (y.size < x.size) -partialCompare(y, x)
  else if (x == y) 0.0
  else Double.NaN
```

It's a bit junkier, but in the worst case you are doing one iteration over the smaller of the two sets instead of potentially three.